### PR TITLE
Add Contract endpoint and queries

### PR DESCRIPTION
**Pull request by gwynndp**
Add endpoint and queries to support contract tool

### CODE REVIEW

**Comment by gwynndp** [comment]
@Kpoke Please review when you have a chance!

File: server/infra/database/ContractRepository.ts [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+              ON grower_account.id = ${this.tableName}.worker_id
+        `,
+        ),
+      )
+      .where((builder) => this.filterWhereBuilder(filter, builder));
+
+    return result[0].count;
+  }
+
+  async getById(id: string | number) {
```
Comment (Kpoke) [inline server/infra/database/ContractRepository.ts:160]: id should just be a string

**Comment by dadiorchen** [comment]
@gwynndp @Kpoke I'm thinking if we put more and more query API crossing schema into this repo, there might be some impact on the domain design, that supposed to be based on microservices?

**Comment by Kpoke** [comment]
It depends, if all information needed can be gotten from the contract API alone then that is fine, but if cross-schema joins are needed then we would have to use the query API.

**Comment by gwynndp** [comment]
@Kpoke @dadiorchen Do we have documentation of all the apps using these services and what data they need?

**Comment by Kpoke** [comment]
Do you mean the query API or the contract service? For the contract service just the admin panel for now, for the query API , I believe it is just the admin panel/web map

**Review by Kpoke — APPROVED** [review]

**Pull request merged**